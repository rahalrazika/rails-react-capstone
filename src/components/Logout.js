import { useHistory } from 'react-router';

const Logout = () => {
  const history = useHistory();

  const handleClick = () => {
    window.localStorage.removeItem('token');
    window.localStorage.removeItem('user');
    history.push('/login');
  };
  return (
    <button
      onClick={handleClick}
      type="button"
      className="p-2 uppercase nav-links "
    >
      Logout
    </button>
  );
};

export default Logout;
