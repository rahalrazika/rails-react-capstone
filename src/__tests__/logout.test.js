import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Logout from '../components/Logout';

it('renders Header  page correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Logout />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

