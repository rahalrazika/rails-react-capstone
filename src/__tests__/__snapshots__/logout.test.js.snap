// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header  page correctly 1`] = `
<button
  className="p-2 uppercase nav-links "
  onClick={[Function]}
  type="button"
>
  Logout
</button>
`;
